kubespawner/objects.py

```diff
@@ -11,13 +33,15 @@ def make_pod_spec(
     run_as_uid,
     fs_gid,
     env,
+    node_selector,
```

Comment:
Can you split out the additional features into separate commits (or even a separate PR) so it's easier to review? I'll leave commnets about them here too tho